ams/optimizer/flink-optimizer/src/main/java/com/netease/arctic/optimizer/flink/FlinkExecutor.java

```diff
@@ -18,26 +18,57 @@
 
 package com.netease.arctic.optimizer.flink;
 
+import static org.apache.flink.configuration.HighAvailabilityOptions.HA_CLUSTER_ID;
+import static org.apache.flink.configuration.TaskManagerOptions.HOST;
+import static org.apache.flink.configuration.TaskManagerOptions.TASK_MANAGER_RESOURCE_ID;
+
 import com.netease.arctic.optimizer.common.OptimizerExecutor;
+import org.apache.flink.metrics.Counter;
 import org.apache.flink.streaming.api.operators.AbstractStreamOperator;
 import org.apache.flink.streaming.api.operators.OneInputStreamOperator;
 import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class FlinkExecutor extends AbstractStreamOperator<Void>
     implements OneInputStreamOperator<String, Void> {
 
+  private static final Logger LOG = LoggerFactory.getLogger(FlinkExecutor.class);
   private final OptimizerExecutor[] allExecutors;
   private OptimizerExecutor executor;
+  private String optimizeGroupName;
 
   public FlinkExecutor(OptimizerExecutor[] allExecutors) {
     this.allExecutors = allExecutors;
   }
 
+  public FlinkExecutor(OptimizerExecutor[] allExecutors, String optimizeGroupName) {
+    this.allExecutors = allExecutors;
+    this.optimizeGroupName = optimizeGroupName;
+  }
+
   @Override
   public void open() throws Exception {
     super.open();
     int subTaskIndex = getRuntimeContext().getIndexOfThisSubtask();
+    String taskManagerContainerId =
```

Comment:
`Container` is a concept in Yarn. Flink has multiple deployment mode(Yarn,k8s). Maybe it is better to call it `taskManagerId`?

Reply:
+1

Reply:
done